File: backend/app/Models/Direccion.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Direccion extends Model
{
    use HasFactory;

    protected $table = 'direcciones';

    protected $fillable = [
        'user_id', 'alias', 'calle', 'ciudad', 'estado',
        'codigo_postal', 'referencia', 'lat', 'lng', 'is_default'
    ];

    protected function casts(): array
    {
        return [
            'is_default' => 'boolean',
            'lat' => 'decimal:8',
            'lng' => 'decimal:8',
        ];
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function scopeDefault($query)
    {
        return $query->where('is_default', true);
    }
}

File: izy_cliente/lib/features/checkout/data/models/direccion.dart
class Direccion {
  final int? id;
  final String calle;
  final String? ciudad;
  final String? estado;
  final String? referencia;
  final double lat;
  final double lng;
  final String? alias;

  Direccion({
    this.id,
    required this.calle,
    this.ciudad,
    this.estado,
    this.referencia,
    required this.lat,
    required this.lng,
    this.alias,
  });

  factory Direccion.fromJson(Map<String, dynamic> json) {
    return Direccion(
      id: json['id'],
      calle: json['calle'],
      ciudad: json['ciudad'],
      estado: json['estado'],
      referencia: json['referencia'],
      lat: (json['lat'] as num).toDouble(),
      lng: (json['lng'] as num).toDouble(),
      alias: json['alias'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'calle': calle,
      'ciudad': ciudad,
      'estado': estado,
      'referencia': referencia,
      'lat': lat,
      'lng': lng,
      'alias': alias,
    };
  }

  String get direccionCompleta {
    final partes = [
      calle,
      if (ciudad != null) ciudad,
      if (estado != null) estado,
    ];
    return partes.join(', ');
  }
}
